src/phoebus_driver.cpp

```diff
@@ -244,6 +317,47 @@ TaskCollection PhoebusDriver::RungeKuttaStage(const int stage) {
 #endif
   }
 
+  if (rad_mocmc_active && stage == integrator->nstages) {
+    TaskRegion &tuning_region = tc.AddRegion(num_independent_task_lists);
+    {
+      particle_resolution.val.resize(1); // total
+      for (int i = 0; i < particle_resolution.val.size(); i++) {
+        particle_resolution.val[i] = 0.;
+      }
+      int reg_dep_id = 0;
+      auto &tl = tuning_region[0];
+
+      auto update_count = tl.AddTask(none, radiation::MOCMCUpdateParticleCount, pmesh,
+                                     &particle_resolution.val);
+
+      tuning_region.AddRegionalDependencies(reg_dep_id, 0, update_count);
+      reg_dep_id++;
+
+      auto start_count_reduce =
+          tl.AddTask(update_count, &AllReduce<std::vector<Real>>::StartReduce,
+                     &particle_resolution, MPI_SUM);
+
+      auto finish_count_reduce =
+          tl.AddTask(start_count_reduce, &AllReduce<std::vector<Real>>::CheckReduce,
+                     &particle_resolution);
+      tuning_region.AddRegionalDependencies(reg_dep_id, 0, finish_count_reduce);
+      reg_dep_id++;
+
+      // Report particle count
+      auto report_count =
+          (Globals::my_rank == 0 ? tl.AddTask(
+                                       finish_count_reduce,
+                                       [](std::vector<Real> *res) {
+                                         std::cout << "MOCMC total particles = "
+                                                   << static_cast<long>((*res)[0])
+                                                   << std::endl;
+                                         return TaskStatus::complete;
+                                       },
+                                       &particle_resolution.val)
+                                 : none);
+    }
+  }
+
```

Comment:
Yeah in principle I would like to break this up I think but a lot of this can also be woven together pretty tightly... I'm not sure what the optimal solution is. We may not be sure what it is until all the code is already written...